api/v1alpha1/dnshealthcheckprobe_types.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2023 The MultiCluster Traffic Controller Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DNSHealthCheckProbeSpec defines the desired state of DNSHealthCheckProbe
+type DNSHealthCheckProbeSpec struct {
+	// Port to connect to the host on. Must be either 80, 443 or 1024-49151
+	// +kubebuilder:validation:XValidation:rule="self in [80, 443] || (self >= 1024 && self <= 49151)",message="Only ports 80, 443, 1024-49151 are allowed"
+	Port *int `json:"port,omitempty"`
+	// Host is the value sent in the host header, to route the request to the correct service
+	// +kubebuilder:validation:Pattern=`^[a-z][a-z0-9\-]+\.([a-z][a-z0-9\-]+\.)*[a-z][a-z0-9\-]+$`
+	Host string `json:"host,omitempty"`
```

Comment:
wondering should we match the Gwapi and call this hostname? https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.Hostname